### PR TITLE
DevTools: Inspect node styles

### PR DESCRIPTION
**Pull request by eerii**
This pr restores the element style inspector and the computed styles list.

**New features:**

- View the applied styles for a node (the ones defined explicitly). This includes the style attribute (for example `<p style="color: blue">`) and the styles defined in stylesheets (like `p { color: blue; }`)
- Modify the applied styles or add a new one. Preview the changes live.
- View the computed styles for a node (all of the calculated css properties).
- Show all of the supported CSS properties, mark the ones that are spelled wrongly and autocomplete names.

**Future work:**

- Disable properties with the checkbox (start with `getCompatibility` for the inspector actor)
- Handle pseudo elements (check `AppliedEntry::pseudo_element` in the page style actor)
- Improve the way it works with big websites, and add some type of caching

image

---
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes are a part of #32404
- [x] These changes do not require tests because there are no tests yet for DevTools

### CODE REVIEW

File: components/devtools/actors/browsing_context.rs [hunk 1]
```diff
@@ -229,7 +229,10 @@ impl BrowsingContextActor {
 
         let accessibility = AccessibilityActor::new(actors.new_name("accessibility"));
 
-        let css_properties = CssPropertiesActor::new(actors.new_name("css-properties"));
+        let (tx, rx) = ipc::channel().unwrap();
+        script_sender.send(GetCssDatabase(tx)).unwrap();
+        let properties = rx.recv().unwrap();
```
Comment (Taym95) [inline components/devtools/actors/browsing_context.rs:234]: should we at least use `expect` with message or print warning for `.send` and `.recv`?

Comment (eerii) [reply components/devtools/actors/browsing_context.rs:234]: Definitely, I was quickly testing that everything worked but it needs some clean up. Thank you for pointing it out!

**Comment by eerii** [comment]
This should be ready for review now @mrobinson :)

File: components/devtools/actors/browsing_context.rs [hunk 2]
```diff
@@ -229,7 +229,13 @@ impl BrowsingContextActor {
 
         let accessibility = AccessibilityActor::new(actors.new_name("accessibility"));
 
-        let css_properties = CssPropertiesActor::new(actors.new_name("css-properties"));
+        let properties = (|| {
+            let (tx, rx) = ipc::channel().ok()?;
```
Comment (mrobinson) [inline components/devtools/actors/browsing_context.rs:233]:
```suggestion
            let (properties_sender, properties_receiver) = ipc::channel().ok()?;
```

File: components/devtools/actors/inspector/page_style.rs [hunk 3]
```diff
@@ -142,29 +122,125 @@ impl Actor for PageStyleActor {
...
+                )
+                .unwrap_or_default()
+                .into_iter()
+                .filter_map(|node| {
+                    let inherited = (node.actor != target).then(|| node.actor.clone());
+                    let node_actor = registry.find::<NodeActor>(&node.actor);
+
+                    // Get the css selectors that match this node present in the currently active stylesheets
+                    let selectors = (|| {
+                        let (tx, rx) = ipc::channel().ok()?;
```
Comment (mrobinson) [inline components/devtools/actors/inspector/page_style.rs:147]:
```suggestion
                        let (selector_sender, selectors_receiver) = ipc::channel().ok()?;
```

File: components/devtools/actors/inspector/page_style.rs [hunk 4]
```diff
@@ -125,13 +103,15 @@ impl Actor for PageStyleActor {
...
-    /// - `getComputed`: Returns the computed styles for a node, placeholder
+    /// - `getComputed`: Returns the computed styles for a node, these include all of the supported
+    /// css properties calculated values.
     ///
-    /// - `getLayout`: Returns the box layout properties for a node, placeholder
+    /// - `getLayout`: Returns the box layout properties for a node.
     ///
-    /// - `isPositionEditable`: Informs whether you can change a style property in the inspector
+    /// - `isPositionEditable`: Informs whether you can change a style property in the inspector.
     fn handle_message(
```
Comment (mrobinson) [inline components/devtools/actors/inspector/page_style.rs:115]: This getting quite large. Would it be possible to split it into methods -- one per message?

File: components/devtools/actors/inspector/style_rule.rs [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+            node,
+            selector,
+        }
+    }
+
+    pub fn applied(&self, registry: &ActorRegistry) -> Option<AppliedRule> {
+        let node = registry.find::<NodeActor>(&self.node);
+        let walker = registry.find::<WalkerActor>(&node.walker);
+
+        let (tx, rx) = ipc::channel().ok()?;
```
Comment (mrobinson) [inline components/devtools/actors/inspector/style_rule.rs:151]:
```suggestion
        let (document_sender, document_receiver) = ipc::channel().ok()?;
```

**Review by mrobinson — APPROVED** [review]

**Comment by mrobinson** [comment]
Great work! I've pushed a new commit with some minor nits, but otherwise this looks good. I'll push to the MQ now.

**Pull request merged**